homeassistant/components/switchgrid/calendar.py

```diff
@@ -0,0 +1,82 @@
+"""Support for Switchgrid Calendar platform."""
+
+import datetime
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN
+from .coordinator import SwitchgridCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the calendar platform for entity."""
+    coordinator: SwitchgridCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([SwitchgridCalendarEntity(coordinator, config_entry)])
+
+
+class SwitchgridCalendarEntity(
+    CoordinatorEntity[SwitchgridCoordinator], CalendarEntity
+):
+    """A calendar entity holding Switchgrid Electric Loadshaving events."""
+
+    entity_id = "calendar.switchgrid_events"
```

Comment:
Happy to get feedback on that point : that's the only way I found to have a name I can use in automations that's always the same no matter what language the end user selects.

Reply:
Just let the name be what it is, people can also change it, breaking it again

Reply:
ok 👌

Reply:
Why did you want a stable entity ID to begin with @licarth ?